src/io.c

```diff
@@ -1269,18 +1239,19 @@ static Char GetLine(void)
     }
 
     /* bump the line number                                                */
-    if (IO()->Input->line < STATE(In) && *(STATE(In) - 1) == '\n') {
+    if (STATE(In)[-1] == '\n') {
```

Comment:
This feels like it has different semantics to before, but I can't find any issues with testing

Reply:
Well, this looks at the previous character, and in the past, that was only possible if the current character did not happen to be the first on the line. Otherwise, we'd perform an out of bound access. But now that byte 0 is reserved for "pushed back" chars after a peek, that's not an issue anymore...

That said, there are all kinds of corner cases to consider (like: what if the pushed back char is a newline? well, it can't be, because we only ever call `PEEK_NEXT_CHAR` if the current char is a dot `.` ; but of course that's not the kind of knowledge that is "obvious" when looking at this code.  Also, in theory, someone might want to use `PEEK_NEXT_CHAR` for another syntax extension; and then whoever did that most likely would not be aware of the need to check this code... Well, perhaps if we left a really good comment which mentions `PEEK_NEXT_CHAR` ? Hrm.

So perhaps the check should indeed be reinstated, just to be on the safe side?

Reply:
I added the check back, but reversed the first condition, as I found it easier to understand this way.